### ISSUE TITLE
iP16, IOS18, Yubikey 5C NFC, Passkey Sign-in?

### ISSUE THREAD

**Issue by thewho2018** [bitwarden/ios#973]
### Bitwarden Beta

- [X] I'm using the new native Bitwarden Beta app and I'm aware that legacy .NET app bugs should be reported in [bitwarden/mobile](https://github.com/bitwarden/mobile)

### Steps To Reproduce

1. Open **YubiKey Manager** go to tab **Interfaces** verify that all boxes under **NFC** are **checked**.

2. Go to **[vault.bitwarden.com -> Settings](https://vault.bitwarden.com/#/settings/security/two-factor)** on a computer and under Providers select **Passkey** add two **YubiKey 5C NFC**, make sure to save on **Security Key** and not fail like i did at first and save it in Windows..

3. On iPhone try to sign-in using **WebAuthn**. "Error has occurred".

Might just be that I'm not understanding every part of this or missing something but i have been emailing back and forth with Ethan at support and i think hes getting frustrated with me!

### Expected Result

Successful sign-in. 

### Actual Result

Error has occurred.

### Build Version

2024.9.2

### Environment Details

iPhone 16 Pro Max. IOS 18.
YubiKey 5C NFC. Firmware 5.2.7
Bitwarden iPhone app 2024.9.2
Yubikey Manager 1.2.6
Yubico Authenticator 7.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

### Expected Result

Successful sign-in.

### Actual Result

Error has occurred.

### Build Version

2024.9.2

### Environment Details

iPhone 16 Pro Max. IOS 18.
YubiKey 5C NFC. Firmware 5.2.7
Bitwarden iPhone app 2024.9.2
Yubikey Manager 1.2.6
Yubico Authenticator 7.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by cbbit** [comment]
Hi there,

Thank you for your report! 

I was able to reproduce this issue, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Thanks once again!

**Comment by hkraal** [comment]
This issue seems to apply to a user/pass login as well (iP16 / iOS 18)

**Comment by SergeantConfused** [comment]
Hello everyone,

Is anyone able to still reproduce this with Bitwarden 2025.1.3?

Thank you in advance,

**Comment by AnthonyWharton** [comment]
I can confirm webauthn logins are working in 2025.2.0, however there are still some cases where this can fail, see: https://github.com/bitwarden/ios/issues/1214

**Comment by dinonovak** [comment]
Iphone 17 Pro, latest iOS, app simply does not recognise NFC Youbikey whatever you try.
Restart, Case on/Off, tapping all around back.

**Comment by pasqualtroncone** [comment]
I still suffering this. iPhone SE. iOS 18.7.2. Bitwarden 2025.12.0 (2787)
It does not recognise NFC Youbikey

**Comment by 3v6bih2w** [comment]
I think I've found a few discussions of similar issues around.  My apologies as I'm not sure how to mention or link other issues (I'm fairly new here and new to security keys)

I'm using an iPhone17 running iOS26.2

Similar thread here: https://github.com/bitwarden/ios/issues/1214
Another thread here: https://github.com/feedback-assistant/reports/issues/626

I have not attempted the Passkey login on Bitwarden yet.  My issue appears to be with using Security Keys / Passkeys / Yubikeys for the 2FA login step using WebAuthn.  I suspect the process is very similar to the Passkey login as I think both use WebAuthn.

My security keys are Token2 FIDO2 NFC PIN+ Release 3.3.

At least in my case, the problem appears to stem from the setting on my keys for Enforce Verification "Always_UV".  When this is turned on, the setting requires a PIN every time the key is used, regardless whether the site requests a PIN or not.  I assume this is the same setting on my Token2 keys as mentioned in the #1214 for Yubikey setting "_ykman fido config toggle-always-uv_".  I think on Yubikey I saw that this setting is an option or is always enabled with Firmware 5.7 and newer, but it can be disabled with a terminal command.

When I have the require PIN setting enabled on my Token2, I have tested using NFC on my iPhone 17 iOS26.2.  The Bitwarden iOS app (version 2025.12.1), Brave Browser iOS (version 1.85), and Safari browser all fail.  I enter my username and password, click "Read Security Key" (browsers) or "Launch WebAuthn" (Bitwarden iOS app), chose "Use Security Key" in the window that pops up, press continue, tap my NFC security key to the phone, enter PIN, it asks me to tap the security key a 2nd time, and after that 2nd NFC tap I get an error message.  The browser error says "authentication cancelled or took to long" and the Bitwarden iOS app seems to output a very long bug report of some sort.

If I turn off the "Always_UV" setting on my Token2 security key, everything works perfectly.  So I am nearly certain this issue is with the always require PIN setting on these security keys.

This same key, with the "Always_UV" require PIN enabled works fine on both the Bitwarden Windows11 app and with Brave and Edge browsers on Windows11.   

So this problem with enabled PIN seems to be iPhone related?  Or Bitwarden-related but only when using iPhone? 

I will experiment with the Always_UV PIN requirement on my iPhone with other services besides Bitwarden.  If the other services seem to work OK with the PIN requirement enabled, then maybe we can determine if this is a Bitwarden issue or an Apple iPhone/iOS issue.

To anyone following this thread or the similar linked threads above, could you please try disabling the Always_UV setting on your security key and report your results?

Bottom line: I'd like to be able to enable Always_UV enforce PIN for the additional security.  I'm just not sure if I need to be working with Bitwarden or if I need to be reporting issues to Apple.

**Comment by 3v6bih2w** [comment]
Follow up to my prior post:   I'm having the same Always_UV Enforce PIN issue on Brave and Safari on iPhone with ProtonMail.  Doesn't work with Enforce PIN, works fine Enforce PIN disabled.  Interestingly the iOS ProtonMail app works fine with the Enforce PIN requirement, only the browser login is a problem.  ProtonMail app and browser logins on Windows11 work fine with Enforced PIN enabled.
